src/gameState.ts

```diff
@@ -0,0 +1,299 @@
+import {Position} from "./position"
+import {Probability} from "./probability"
+
+export class GameState{
+    public ShipPositions:Array<any>;
+    public MyShots:Array<any>;
+    public OpponentsShots:Array<any>;
+    public remainingShips:number[] = [2,3,3,4,5];  
+    // note that remainingShips always contains the shiplengths in ascending order
+    public getBoard():number[][]{
+        return this.board;
+    }
+    public static letterToNumber:any = {
+        "A":0, 
+        "B":1, 
+        "C":2, 
+        "D":3,
+        "E":4,
+        "F":5,
+        "G":6,
+        "H":7,
+        "I":8,
+        "J":9
+    };
+    public static numberToLetter:any = {
+        0:"A",
+        1:"B",
+        2:"C",
+        3:"D",
+        4:"E",
+        5:"F",
+        6:"G",
+        7:"H",
+        8:"I",
+        9:"J"
+    };
+
+    // 0 means unchecked, 1 means hit,2 means sunk, -1 means missed
+    public board:number[][] = [
+        [0,0,0,0,0,0,0,0,0,0],
+        [0,0,0,0,0,0,0,0,0,0],
+        [0,0,0,0,0,0,0,0,0,0],
+        [0,0,0,0,0,0,0,0,0,0],
+        [0,0,0,0,0,0,0,0,0,0],
+        [0,0,0,0,0,0,0,0,0,0],
+        [0,0,0,0,0,0,0,0,0,0],
+        [0,0,0,0,0,0,0,0,0,0],
+        [0,0,0,0,0,0,0,0,0,0],
+        [0,0,0,0,0,0,0,0,0,0]
+    ];
+
+    // reqBody is the request body, a json object
+    constructor(reqBody:any){
```

Comment:
As a general rule, to reduce confusion when someone opens a new class, you would want to order your class into:

1. private class variables 
2. public class variables
3. constructor
4. public methods
5. private methods

You can go off and do your own thing, but you must be aware that it makes it harder for people to follow (as it wont be in the order they expect), so should have a good reason for doing so.